charts/redpanda/chart_test.go

```diff
@@ -238,6 +239,9 @@ func extractRedpandaConfigsFromConfigMap(manifests []byte) (*configmapRepresenta
 				}
 				result.redpanda = string(jsonR)
 
+				rex := regexp.MustCompile("\"overprovisioned\":(true|false)")
```

Comment:
nit: This is kinda hidden right now and using regex to change up JSON always feels a bit sketchy. What about moving this up to line 205 and making it something like:

```go
// If this path is deeper, use the `unstructured` package from k8s to access it
// Explain the old bug here. Now that it's considered fixed, we'll ignore this diff by pushing the correct value into the old conf
oldConf.redpanda["overprovisioned"] = newConf.redpanda["overprovisioned"]
```

Placing this right before the assertion will help group up all the logic that we start to accumulate like this.

You could also compute a JSON diff and remove the `overprovisioned` field from it in a similar way.

Reply:
> What about moving this up to line 205
>
> If this path is deeper, use the `unstructured` package from k8s to access it

I can not use on manifests as `---` (YAML delimiter) is unrecognized by the unstructured unmarshall function.

> Placing this right before the assertion will help group up all the logic that we start to accumulate like this.

I agree.

> You could also compute a JSON diff and remove the overprovisioned field from it in a similar way.

Which JSON diff function/library?

Reply:
> I can not use on manifests as --- (YAML delimiter) is unrecognized by the unstructured unmarshall function.

Wouldn't that be a non issue around line 205 as you won't be working with manifests (multi-document YAMLs) but with the parsed out configmap values?

Reply:
`newConf.redpanda` holds a raw redpanda config not K8S ConfigMap. When I'm unmarshall raw redpanda config I get the error:
```
Received unexpected error:
Object 'Kind' is missing in ...
```